src/megaclient.cpp

```diff
@@ -2025,11 +2069,12 @@ void MegaClient::exec()
                 break;
 
             case REQ_INFLIGHT:
-                if (Waiter::ds >= (pendingsc->lastdata + HttpIO::SCREQUESTTIMEOUT))
+                if (!pendingscTimedOut && Waiter::ds >= (pendingsc->lastdata + HttpIO::SCREQUESTTIMEOUT))
                 {
                     LOG_debug << "sc timeout expired";
-                    delete pendingsc;
-                    pendingsc = NULL;
+                    // In almost all cases the server won't take more than 40 seconds.  But if it does, break the cycle of endless requests for the same thing
```

Comment:
```suggestion
                    // In almost all cases the server won't take more than SCREQUESTTIMEOUT seconds.  But if it does, break the cycle of endless requests for the same thing
```
Otherwise, in case we change it in the future, the `40` might be misleading :)

Reply:
updated